### ISSUE TITLE
Support parallel execution of legacy Junit Tests

### ISSUE THREAD

**Issue by ravikulkarni** [junit-team/junit5#515]
Requesting to add support for parallel execution of legacy JUnit tests.

**Use case:**
- Ability to execute legacy JUnit tests in parallel  - Legacy tests are implemented in older versions of JUnit. The tests and their helper classes have shared static and singleton  variables all over the place. The classes need not be modified before running (or could not be modified because there are lots of test cases to refactor)

[How it is implemented with JUnit 4 - Use custom test class loader & threads to implement this. I will later share my implementation with Junit4]

**Comment by marcphilipp** [comment]
Parallel support is being discussed in #60. Sounds like you'd need to provide a custom class loader, too. Can you show us how you implemented this using JUnit 4?

**Comment by ravikulkarni** [comment]
Here is an example of how I have implemented this. [Github Link](https://github.com/ravikulkarni/parallel_junit)

I was considering implementing this in JUnit 5 using Launcher APIs. Thoughts?

**Comment by marcphilipp** [comment]
Duplicate of #2229
